library/common.h

```diff
@@ -169,6 +169,24 @@ inline void mbedtls_xor(unsigned char *r, const unsigned char *a, const unsigned
 #endif
 /* *INDENT-ON* */
 
+/*
+ * Define the constraint used for pointer operands to asm.
+ *
+ * This is normally the usual "r", but for aarch64_32 (aka ILP32,
+ * as found in watchos), "p" is required to avoid warnings from clang.
```

Comment:
Not just warnings, and not just clang: on ILP32 (more precisely, the P32 part), a pointer is 32 bits, but on aarch64, `r` means 64 bits.

Why not use `p` all the time? It's correct on all architectures. It's the right thing to use for a memory address.

Reply:
Indeed, using 'p' always was attempt #2 (04cb9ac59ea0943b298ccb1f3db3e4d97da40dff) but it failed the CI. I can't remember exactly why. I'll push it again and see what the issue was.

Reply:
Ah, it's not supported by armclang.